packages/contracts-bridge/contracts/test/BridgeRouter.t.sol

```diff
@@ -1,53 +1,332 @@
+    function test_handleRevertsIfNotCalledByReplica() public {
+        uint32 nonce = 23;
+        bytes memory message = "lol";
+        bytes32 sender = remoteBridgeRouter;
+        vm.expectRevert("!replica");
+        bridgeRouter.handle(remoteDomain, nonce, sender, message);
+    }
+
+    function test_handleRevertsIfSenderNotRegisteredRouter() public {
+        uint32 nonce = 23;
+        bytes memory message = "lol";
+        bytes32 sender = address(0xBEEF).addressToBytes32();
+        vm.prank(mockReplica);
+        vm.expectRevert("!remote router");
+        bridgeRouter.handle(remoteDomain, nonce, sender, message);
+    }
+
+    // The only registured remote router is for domain = remoteDomain and sender = remoteBridgeRouter
+    function testFuzz_handleRevertsIfSenderNotRegisteredRouter(
+        bytes32 sender,
+        uint32 domain
+    ) public {
+        vm.assume(domain != remoteDomain && sender != remoteBridgeRouter);
+        uint32 nonce = 23;
+        bytes memory message = "lol";
+        vm.prank(mockReplica);
+        vm.expectRevert("!remote router");
+        bridgeRouter.handle(domain, nonce, sender, message);
+    }
+
+    function test_handleRevertsIfSenderNotCorrectDomain() public {
+        uint32 nonce = 23;
+        bytes memory message = "lol";
+        bytes32 sender = remoteBridgeRouter;
+        vm.prank(mockReplica);
+        vm.expectRevert("!remote router");
+        bridgeRouter.handle(123, nonce, sender, message);
+    }
+
+    function test_handleRevertsIfMalformedMessage() public {
+        uint32 nonce = 23;
+        bytes memory message = "lol";
+        bytes32 sender = remoteBridgeRouter;
+        vm.prank(mockReplica);
+        vm.expectRevert("Validity assertion failed");
+        bridgeRouter.handle(remoteDomain, nonce, sender, message);
+    }
+
+    function test_handleRevertsIfInvalidAction() public {
+        uint32 nonce = 23;
+        uint256 tokenAmount = 100;
+        bytes32 tokenDetailsHash = "sdf";
+        // Invalid
+        bytes memory action = abi.encodePacked(
+            BridgeMessage.Types.Invalid,
+            tokenReceiver,
+            tokenAmount,
+            tokenDetailsHash
+        );
+        bytes32 id = address(localToken).addressToBytes32();
+        bytes memory tokenId = abi.encodePacked(remoteDomain, id);
+        bytes32 sender = remoteBridgeRouter;
+        bytes memory message = abi.encodePacked(tokenId, action);
+        vm.prank(mockReplica);
+        vm.expectRevert("!valid action");
+        bridgeRouter.handle(remoteDomain, nonce, sender, message);
+    }
+
+    /// @notice Test that the only valid actions are the enums Message.Types.TransferToHook
+    /// and Message.Types.Message, which in numbers are 5 and 3 respectively.
+    /// If the enumAction was a larger number, we get the Validity assertion failed error,
+    /// as the message is not of the correct form
+    function test_handleRevertsIfInvalidAction(uint8 enumAction) public {
```

Comment:
should be `testFuzz` ?

Reply:
speaking of which I think `test_fuzz_` would be better naming convention? but that's kinda a nit

Reply:
We have kinda aligned to this convention so I want to try it first and see how it works
https://book.getfoundry.sh/tutorials/best-practices

Reply:
that's sad. camel case is significantly harder to read

Reply:
Feel free to open a PR and share your pov , this is very much a WIP from  Matt (Scopelift) and Lucas (Maple) and me